### DISCUSSION TITLE
alarm clock based on input_datetime

### DISCUSSION THREAD

**Discussion by phsdv** [custom-components/pyscript#547] [Q&A] [answered]
I am trying to make a light to switch on based on the time stored in a input_datetime helper. I came up with below code, but it never triggers at the time in the  input_datetime helper.
What am I doing wrong?
```
@state_trigger("input_datetime.alarm_time")
@time_trigger("once(now)")
def update_alarm():
    log.info(f"time has changed, updating alarm to {input_datetime.alarm_time}")

    @time_trigger(f"once({input_datetime.alarm_time})")
    def alarm():
        log.info("New alarm time set")
        light.turn_on("light.someroom")
```

**Comment by craigbarratt** [comment, marked as answer]
First, have you confirmed the string `f"once({input_datetime.alarm_time})"` produces a correctly formatted time?

The main problem is that the `alarm()` function is destroyed as soon as `update_alarm()` returns, since there is no longer any reference to it in any scope, so there will no longer be any trigger.  See the [docs](https://hacs-pyscript.readthedocs.io/en/latest/reference.html#trigger-closures).  You need to keep some reference to the closure function so it remains defined after the enclosing function returns, eg, by setting a global variable.

**Reply by phsdv** [reply]
that makes sense, thanks for explaining.
